nqeit-iceberg-dremio-extension/src/main/java/org/projectnessie/integtests/dremio/DremioHelper.java

```diff
@@ -114,17 +114,40 @@ private String performHttpRequest(String url, String payload) throws IOException
     return readResponse(con, url);
   }
 
+  private static void sleep(Duration duration) {
+    try {
+      Thread.sleep(duration.toMillis());
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+    }
+  }
+
   private void waitForJobCompletion(String jobId, String query) throws IOException {
-    // The doc for getting the job status for cloud is not there, but it is similar to software
-    // See docs: https://docs.dremio.com/software/rest-api/jobs/get-job/
+    // See docs: https://docs.dremio.com/cloud/reference/api/job/
     String url = getJobUrl(jobId);
     Set<String> finalJobStates = new HashSet<>(asList("COMPLETED", "FAILED", "CANCELED"));
     // Default Timeout for engine-startup is 5min
     Duration timeout = Duration.ofMinutes(5);
     Instant deadline = Instant.now().plus(timeout);
 
+    boolean first = true;
     while (true) {
-      String responseBody = performHttpRequest(url, null);
+      if (first) {
+        first = false;
+      } else {
+        sleep(Duration.ofSeconds(5));
+      }
+      String responseBody;
+      try {
+        responseBody = performHttpRequest(url, null);
+      } catch (IOException e) {
+        boolean rateLimited = e.getMessage() != null && e.getMessage().contains("http code: 429");
+        if (rateLimited) {
+          // retry
+          continue;
+        }
+        throw e;
+      }
       JsonNode node = parseJson(responseBody, url);
       String jobState = node.get("jobState").textValue();
       if (finalJobStates.contains(jobState)) {
```

Comment:
I wonder, whether it makes sense to check for `ENGINE_START` here?

Reply:
imo by polling every 5 sec we are already avoiding getting rate limited by factor 5, which is good enough until we have contrary data.

usually when orgs/projects are created from scratch (as we do in CI) the engines are pre-warmed, so this isnt likely to happen very frequently.
the 5min timeout was originally added because tests were run locally against a pre-created org/project (whose enginse went offline after sitting idle for too long).

also in theory jobs can be in QUEUED state when engines are at their concurrency limit, which would then mean polling every 1 sec is overkill yet again.
i'd say we stick to 5sec universally and see how it goes.
on average job completion is noticed 2,5sec later than it happens which is fine for our tests?